### PR TITLE
Add error handling section

### PR DESCRIPTION
**Pull request by hoseinz3**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,6 +34,9 @@
   5. [Objects and Data Structures](#objects-and-data-structures)
      * [Use object encapsulation](#use-object-encapsulation)
      * [Make objects have private/protected members](#make-objects-have-privateprotected-members)
+  6. [Error handling](#error-handling)
+     * [Use Exceptions Rather Than Return Codes](#use-exceptions-rather-than-return-codes)
+     * [Don't Return Null](#dont-return-null)
   6. [Classes](#classes)
```
Comment (Jibbarth) [inline README.md]: Even if Markdown replace correctly number in ordered list, we should update the other numbers for lisibility ?

Comment (hoseinz3) [reply README.md]: @Jibbarth i don't know, but fixed for consistency.

**Comment by hoseinz3** [comment]
@TomasVotruba let me know if this PR has an issue ;)

File: README.md [hunk 2]
```diff
@@ -1247,6 +1250,140 @@ echo 'Employee name: '.$employee->getName(); // Employee name: John Doe
...
+        $this->retriveDeviceRecord($handle);
+        $this->pauseDevice($handle);
+        $this->clearDeviceWorkQueu();
+        
+    }
+
+    private function getHandle($devicdId): DeviceHandle
+    {
+        $handle = new DeviceHandle($devicdId);
+        if ($handle == DeviceHandle::INVALID) {
```
Comment (peter-gribanov) [inline README.md:1324]: Are you sure that this is good?

**Comment by TomasVotruba** [comment]
I didn't get the notification, will get to it :+1:

File: README.md [hunk 3]
```diff
@@ -1247,6 +1250,140 @@ echo 'Employee name: '.$employee->getName(); // Employee name: John Doe
...
+    private function getHandle($deviceId): DeviceHandle
+    {
+        return new DeviceHandle($devicdId);
+    }
+
+    private function retriveDeviceRecord(DeviceHandle $handle): DeviceRecord
+    {
+        return new DeviceRecord($handle);
+    }
+}
```
Comment (TomasVotruba) [inline README.md:1292]: There is too much going on in this example. Shutting down, factories, logging, record validation.
I can't even find the *return code* part.

Example with as less code as possible expressing only the point you try to explain would be better.

File: README.md [hunk 4]
```diff
@@ -1247,6 +1250,140 @@ echo 'Employee name: '.$employee->getName(); // Employee name: John Doe
...
+            return $totalPay += $employee->getPay();
+        }, 0);
+    }
+    return 0;
+}
+
+public function getEmployees(): array
+{
+    return $this->employees;
+}
```
Comment (TomasVotruba) [inline README.md:1364]: Here `getEmployees(): array` returns always array.
It should be maybe `getEmployees(): ?array`?

File: README.md [hunk 5]
```diff
@@ -1247,6 +1250,140 @@ echo 'Employee name: '.$employee->getName(); // Employee name: John Doe
...
+    $employees = $this->getEmployees();
+    return array_reduce($employees, function($totalPay, $employee) {
+        return $totalPay += $employee->getPay();
+    }, 0);
+
+}
+
+public function getEmployees(): array
+{
+    return is_null($this->employees) ? [] : $this->employees;
```
Comment (TomasVotruba) [inline README.md:1381]: What about including `$empoyees` property and set it default value of `[]`?

**Review by TomasVotruba — COMMENTED** [review]
I like the idea and added my review

File: README.md (hunk 3 above)
Comment (hoseinz3) [reply README.md:1292]: @TomasVotruba i tried to use [clean code](https://www.amazon.com/Clean-Code-Handbook-Software-Craftsmanship/dp/0132350882) book's example in error handling section, but i agree with you, i will try to show error handling in another way with simple code.

File: README.md [hunk 6]
```diff
@@ -1247,6 +1250,140 @@ echo 'Employee name: '.$employee->getName(); // Employee name: John Doe
...
+class DeviceController 
+{
+    const DEV1 = "/dev/dev1";
+    private $record;
+
+    public function sendShutDown(): void
+    {
+        $handle = $this->getHandle(self::DEV1);
+
+        if ($handle != DeviceHandle::INVALID) {
```
Comment (peter-gribanov) [inline README.md:1269]: Same as [here](https://github.com/jupeter/clean-code-php/pull/138/files#r169024432). This condition will always be [`TRUE`](https://3v4l.org/4IDTa).

**Review by peter-gribanov — CHANGES_REQUESTED** [review]

**Comment by TomasVotruba** [comment]
Any plan to move with this or is that to be closed?

**Comment by TomasVotruba** [comment]
Closing for no response from the author for almost a 3/4 year

**Pull request closed without merging**